### ISSUE TITLE
MinChar interferes with clearing the input

### ISSUE THREAD

**Issue by lukasoppermann** [Haroenv/holmes#50]
When a `minCharacters` option is set, the result is not refreshed when the input is empty.

**Comment by Haroenv** [comment]
Aha, that's a good observation. This can be fixed by disabling the `minCharacters` check on "initialisation" or reset. You are talking about the `h.clear()`, right?

**Comment by lukasoppermann** [comment]
Actually no, I am talking about deleting everything from the input by keyboard or by clicking the **(X)** button that some browser show by default to clear the input.

Maybe you can detect it using the `input` event? In any case, there is also a `reset` event which is triggered. But probably the easiest would be to trigger a search if the input is empty or has more chars than defined in `minCharacters`.

**Comment by Haroenv** [comment]
Okay that makes sense, that was a downside I saw on `minCharacters`. So then `minCharacters` should only run if the length is longer than before, so it would only run in front direction. But then it _would_ get called like this:

```
// minCharacters: 3
//current:
// not called
a // not called
aa // not called
aaa // not called
aaaa // called
aaa // not called
aa // not called
a // not called
// not called
a // not called

//new:
// not called
a // not called
aa // not called
aaa // not called
aaaa // called
aaa // called
aa // called
a // called
// called
a // not called
```

What do you think about that?

**Comment by lukasoppermann** [comment]
Wouldn't it be like this:

```
a // not called
aa // not called
aaa // not called
aaaa // called
aaa // called
aa // not called
a // not called
// called
a // not called
```

**Comment by Haroenv** [comment]
ok, so you want an exception for 0 characters then?

**Comment by lukasoppermann** [comment]
Exactly. This is a UX question for me. As a user I expect that when I empty the input, the search resets. Of course I can add it myself, but I think it would be a very sensible thing to add. 

If you think it is not a use-case for everybody it would be an option you can disable `restOnEmpty = false`. Although I current do not see where it could cause harm.

**Comment by Haroenv** [comment]
Okay, makes sense. I wasn't 100% Sure either, adding it now :)

**Comment by Haroenv** [comment]
This is now available in `v1.11.1`

**Comment by lukasoppermann** [comment]
Tested, works like a charm.
